File: 0057-insert-interval/0057-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        stack<vector<int>>st;
        bool isInserted=false;
        int n=intervals.size();
        for(int i=n-1;i>=0;i--)
        {
            if(intervals[i][0]>newInterval[0])
            {
                st.push(intervals[i]);
            }
            else
            {
                if(!isInserted)
                {
                    st.push(newInterval);
                    isInserted=true;
                }
                st.push(intervals[i]);
            }
        }
        if(!isInserted)
            st.push(newInterval);
        
        vector<vector<int>>result;
        while(st.size()!=1)
        {
            vector<int>itv1=st.top();
            st.pop();
            vector<int>itv2=st.top();
            st.pop();
            if(itv1[1]<itv2[0])
            {
                result.push_back(itv1);
                st.push(itv2);
            }
            else
            {
                st.push({min(itv1[0],itv2[0]),max(itv2[1],itv1[1])});
            }
        }
        result.push_back(st.top());
        return result;
    }
};

File: 0057-insert-interval/0057-insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        List<int[]> temp = new ArrayList<>();
        int pt=0,strt=0;
        boolean isAdded=false;
        if(intervals.length>0&&newInterval[0]>intervals[0][1])
        {
            strt=1;
            temp.add(intervals[0]);
        }
        else
        {
            isAdded=true;
            temp.add(newInterval);
        }
        for(int i=strt;i<intervals.length;i++)
        {
            if(temp.get(pt)[1]>=intervals[i][0])
            {
                temp.get(pt)[0] = Math.min(temp.get(pt)[0], intervals[i][0]);
                temp.get(pt)[1] = Math.max(temp.get(pt)[1], intervals[i][1]);
            }
            else
            {
                if(!isAdded && (newInterval[0]<=intervals[i][1]))
                {
                    temp.add(newInterval);
                    isAdded=true;
                    pt++;
                    i--;
                    continue;
                }
                temp.add(intervals[i]);
                pt++;
            }
        }
        if(!isAdded)
        {
            if(newInterval[0]<=temp.get(pt)[1])
            {
                temp.get(pt)[0] = Math.min(temp.get(pt)[0], newInterval[0]);
                temp.get(pt)[1] = Math.max(temp.get(pt)[1], newInterval[1]);
            }
            else
            {
                temp.add(newInterval);
            }
        }

        return temp.toArray(new int[0][]);
    }
}